cocotb/drivers/amba.py

```diff
@@ -112,12 +112,16 @@ def _send_write_data(self, data, delay=0, byte_enable=0xF):
 
     @cocotb.coroutine
     def write(self, address, value, byte_enable=0xf, address_latency=0,
-              data_latency=0):
+              data_latency=0, sync=True):
         """
         Write a value to an address.
 
         The *_latency KWargs allow control over the delta
+
+        sync dictates whether we wait for a clock edge or not
         """
+        if sync:
+            yield RisingEdge(self.clock)
```

Comment:
It's not clear what the motivation for this change is - the commit message doesn't say anything about it.

Reply:
In the docstring it states that it's for waiting for a clock edge - synchronising the write with a clock edge

Reply:
To be clear, I'm asking "why should this exist", not "what does this do".

In particular, I worry that the behavior of this might be confusing now:
```python
yield RisingEdge(self.clock)
yield obj.write(..., sync=True)
```
The caller might be surprised to find that despite `write` being called at a point where it can behave "synchronously", passing `sync=True` causes another cycle to pass.